plugins/module_utils/cmd_runner.py

```diff
@@ -261,10 +263,11 @@ def run(self, **kwargs):
         for arg_name in self.args_order:
             value = None
             try:
-                value = named_args[arg_name]
+                if arg_name in named_args:
+                    value = named_args[arg_name]
+                elif not runner.arg_formats[arg_name].ignore_missing_value:
+                    raise MissingArgumentValue(self.args_order, arg_name)
```

Comment:
This will be caught by the below `except Exception` and converted to a `FormatError` - which is why the tests fail.

The reason why the tests pass **in some cases** is that with many newer Python versions, exceptions that are raised while handling an exception handler are kept with the exception and also printed when the module crashes.

So the module crashes (instead of failing!) when the tests pass with stderr
```
Traceback (most recent call last):
  File "/tmp/ansible_cmd_echo_payload_n_xmegqp/ansible_cmd_echo_payload.zip/ansible_collections/community/general/plugins/module_utils/cmd_runner.py", line 269, in run
ansible_collections.community.general.plugins.module_utils.cmd_runner.MissingArgumentValue: Cannot find value for parameter bb in ('aa', 'bb')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 133, in <module>
  File "<stdin>", line 125, in _ansiballz_main
  File "<stdin>", line 73, in invoke_module
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 209, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_cmd_echo_payload_n_xmegqp/ansible_cmd_echo_payload.zip/ansible/modules/cmd_echo.py", line 55, in <module>
  File "/tmp/ansible_cmd_echo_payload_n_xmegqp/ansible_cmd_echo_payload.zip/ansible/modules/cmd_echo.py", line 46, in main
  File "/tmp/ansible_cmd_echo_payload_n_xmegqp/ansible_cmd_echo_payload.zip/ansible_collections/community/general/plugins/module_utils/cmd_runner.py", line 272, in run
ansible_collections.community.general.plugins.module_utils.cmd_runner.FormatError: Failed to format parameter bb with value None: Cannot find value for parameter bb in ('aa', 'bb')
```
and when the tests fail, stderr is just
```
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1668480475.55-8125-111677881832624/AnsiballZ_cmd_echo.py", line 126, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-tmp-1668480475.55-8125-111677881832624/AnsiballZ_cmd_echo.py", line 118, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-tmp-1668480475.55-8125-111677881832624/AnsiballZ_cmd_echo.py", line 67, in invoke_module
    run_name='__main__', alter_sys=True)
  File "/usr/lib64/python2.7/runpy.py", line 176, in run_module
    fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 82, in _run_module_code
    mod_name, mod_fname, mod_loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/tmp/ansible_cmd_echo_payload_sCMqmF/ansible_cmd_echo_payload.zip/ansible/modules/cmd_echo.py", line 55, in <module>
  File "/tmp/ansible_cmd_echo_payload_sCMqmF/ansible_cmd_echo_payload.zip/ansible/modules/cmd_echo.py", line 46, in main
  File "/tmp/ansible_cmd_echo_payload_sCMqmF/ansible_cmd_echo_payload.zip/ansible_collections/community/general/plugins/module_utils/cmd_runner.py", line 272, in run
ansible_collections.community.general.plugins.module_utils.cmd_runner.FormatError: Failed to format parameter bb with value None: Cannot find value for parameter bb in ('aa', 'bb')
```
Since the tests look for `MissingArgumentValue: Cannot find value for parameter bb`, which is *part of the exception that was handled while the new exception got raised* and thus gets printed with new enough Python versions when crashing, the tests pass.

So here the tests are incomplete and should always fail, instead of just in a few cases.

Reply:
Wow. Nice catch! :-) It seems obvious now that it's been explained. The ``MissingArgumentValue`` was supposed to emerge from the code when that type of problem happens, so instead of adapting the tests I am re-raising that particular exception (instead of raising a `FormatError`.

Reply:
... and problem disappeared! :-)